### PR TITLE
Use manifests images for operator

### PR DESCRIPTION
**Pull request by mhmxs**
This change replaces URL based operator manifests download to job execution with versioned manifests image.

### CODE REVIEW

File: pkg/installer/installer.go [hunk 1]
```diff
@@ -89,227 +80,33 @@ type Installer struct {
...
-	stosClusterFiles, err := createFileData(config.Spec.Install.StorageOSClusterYaml, pluginversion.ClusterLatestSupportedURL(), stosClusterFile)
-	if err != nil {
-		return fs, err
-	}
-
-	// append storageos secret yaml to cluster yaml if necessary. This will happen in the event of an
-	// uninstall of storageos version < 2.5.0.
-	if config.InstallerMeta.StorageOSSecretYaml != "" {
-		stosSecretYaml, err := pullManifest(config.InstallerMeta.StorageOSSecretYaml)
+	if ensureNamespace {
```
Comment (nolancon) [inline pkg/installer/installer.go:91]: I'm not sure why this `ensureNamespace` functionality is necessary.

If the user passes a storageos operator namespace to the CLI, the namespace will be set in the operator's kustomization file here https://github.com/storageos/kubectl-storageos/blob/186f0b2ba372d4836cb7352fbbf4ec7a2c5d329b/pkg/installer/install.go#L121 and kustomized/applied here (creating the namespace if it does not already exist) https://github.com/storageos/kubectl-storageos/blob/186f0b2ba372d4836cb7352fbbf4ec7a2c5d329b/pkg/installer/install.go#L171
Am I missing something?

File: pkg/utils/k8s.go [hunk 2]
```diff
@@ -337,10 +381,130 @@ func GetEtcdCluster(config *rest.Config, name, namespace string) (*etcdoperatora
...
-		if errors.IsNotFound(err) {
+		if kerrors.IsNotFound(err) {
 			return nil
 		}
 		return err
 	}
 	return fmt.Errorf("etcdcluster exists")
 }
+
+// EnsureNamespace Creates namespace if not exists.
```
Comment (nolancon) [inline pkg/utils/k8s.go:392]: Nit: if not exists -> if it does not exist

File: pkg/utils/k8s.go [hunk 3]
```diff
@@ -337,10 +381,130 @@ func GetEtcdCluster(config *rest.Config, name, namespace string) (*etcdoperatora
...
+		return NamespaceExists(config, name)
+	}, 120, 5)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func CreateJobAndFetchResult(config *rest.Config, name, namespace, image string) (string, error) {
```
Comment (nolancon) [inline pkg/utils/k8s.go:423]: Needs description comment

File: pkg/installer/installer.go (hunk 1 above)
Comment (mhmxs) [reply pkg/installer/installer.go:91]: Yes, but job needed to fetch manifest and namespace needed to spin up the job.

Comment (nolancon) [reply pkg/installer/installer.go:91]: Ah I see.. Chicken and egg! 🐔🥚😄

**Review by nolancon — APPROVED** [review]

**Pull request merged**